services/venice-server/src/main/java/com/linkedin/venice/server/VeniceServer.java

```diff
@@ -355,6 +355,11 @@ private List<AbstractVeniceService> createServices() {
       return helixData;
     });
 
+    managerFuture.thenApply(manager -> {
+      storageService.checkWhetherStoragePartitionsShouldBeKeptOrNot(manager);
+      return true;
+    });
+
```

Comment:
What if Helix couldn't compute the new Ideal state quick enough after the host is restarted and connected? Server might delete everything and rebootstrap from scratch?

Reply:
So a restarted host shouldn't alter the idealstate.  idealstate will only be altered after a rebalance has occurred, which will only happen in cases where the node was down for an extended period.